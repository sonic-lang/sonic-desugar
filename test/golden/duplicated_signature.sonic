a :: Int ;;
a = 1 ;;

a :: Int
