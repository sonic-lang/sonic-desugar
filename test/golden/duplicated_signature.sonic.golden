( Module
    { attrs = []
    , bindings =
        [ WithProv
            ( Source ( Generated desugar ) )
            ( BindGroup
                [ WithProv
                    ( Derived desugar
                        ( Derived desugar
                            ( Source
                                ( Parsed
                                    ( Location
                                        { file = "test/golden/duplicated_signature.sonic"
                                        , span = Span
                                            { begin = Position
                                                { line = Line 2
                                                , column = Column 1
                                                }
                                            , end = Position
                                                { line = Line 2
                                                , column = Column 7
                                                }
                                            }
                                        }
                                    )
                                )
                            )
                        )
                    )
                    ( Bind
                        { attrs = WithProv
                            ( Derived desugar
                                ( Source ( Generated desugar ) )
                            )
                            ( Attrs ( fromList [] ) )
                        , id = WithProv
                            ( Source
                                ( Parsed
                                    ( Location
                                        { file = "test/golden/duplicated_signature.sonic"
                                        , span = Span
                                            { begin = Position
                                                { line = Line 2
                                                , column = Column 1
                                                }
                                            , end = Position
                                                { line = Line 2
                                                , column = Column 3
                                                }
                                            }
                                        }
                                    )
                                )
                            ) ( Name "a" )
                        , type_ = Just
                            ( WithProv
                                ( Source
                                    ( Parsed
                                        ( Location
                                            { file = "test/golden/duplicated_signature.sonic"
                                            , span = Span
                                                { begin = Position
                                                    { line = Line 1
                                                    , column = Column 6
                                                    }
                                                , end = Position
                                                    { line = Line 1
                                                    , column = Column 10
                                                    }
                                                }
                                            }
                                        )
                                    )
                                )
                                ( Ctor ()
                                    ( WithProv
                                        ( Source
                                            ( Parsed
                                                ( Location
                                                    { file = "test/golden/duplicated_signature.sonic"
                                                    , span = Span
                                                        { begin = Position
                                                            { line = Line 1
                                                            , column = Column 6
                                                            }
                                                        , end = Position
                                                            { line = Line 1
                                                            , column = Column 10
                                                            }
                                                        }
                                                    }
                                                )
                                            )
                                        )
                                        ( Path
                                            { prefix = Nothing
                                            , inner = SimplePath
                                                { parent = Nothing
                                                , name = Name "Int"
                                                }
                                            }
                                        )
                                    )
                                )
                            )
                        , rhs = WithProv
                            ( Source ( Generated desugar ) )
                            ( Case ()
                                ( WithProv
                                    ( Source ( Generated desugar ) )
                                    ( Var ()
                                        ( WithProv
                                            ( Source ( Generated desugar ) )
                                            ( Path
                                                { prefix = Just Local
                                                , inner = SimplePath
                                                    { parent = Nothing
                                                    , name = Unique ( MkUnique 0 )
                                                    }
                                                }
                                            )
                                        )
                                    )
                                )
                                ( WithProv
                                    ( Source ( Generated desugar ) )
                                    [ WithProv
                                        ( Source ( Generated desugar ) )
                                        ( CaseArm
                                            { pat = WithProv
                                                ( Derived desugar
                                                    ( Source
                                                        ( Parsed
                                                            ( Location
                                                                { file = "test/golden/duplicated_signature.sonic"
                                                                , span = Span
                                                                    { begin = Position
                                                                        { line = Line 2
                                                                        , column = Column 1
                                                                        }
                                                                    , end = Position
                                                                        { line = Line 2
                                                                        , column = Column 3
                                                                        }
                                                                    }
                                                                }
                                                            )
                                                        )
                                                    )
                                                )
                                                ( Var ()
                                                    ( WithProv
                                                        ( Source
                                                            ( Parsed
                                                                ( Location
                                                                    { file = "test/golden/duplicated_signature.sonic"
                                                                    , span = Span
                                                                        { begin = Position
                                                                            { line = Line 2
                                                                            , column = Column 1
                                                                            }
                                                                        , end = Position
                                                                            { line = Line 2
                                                                            , column = Column 3
                                                                            }
                                                                        }
                                                                    }
                                                                )
                                                            )
                                                        ) ( Name "a" )
                                                    )
                                                )
                                            , guard = Nothing
                                            , body = WithProv
                                                ( Source ( Generated desugar ) )
                                                ( Var ()
                                                    ( WithProv
                                                        ( Source
                                                            ( Parsed
                                                                ( Location
                                                                    { file = "test/golden/duplicated_signature.sonic"
                                                                    , span = Span
                                                                        { begin = Position
                                                                            { line = Line 2
                                                                            , column = Column 1
                                                                            }
                                                                        , end = Position
                                                                            { line = Line 2
                                                                            , column = Column 3
                                                                            }
                                                                        }
                                                                    }
                                                                )
                                                            )
                                                        )
                                                        ( Path
                                                            { prefix = Just Local
                                                            , inner = SimplePath
                                                                { parent = Nothing
                                                                , name = Name "a"
                                                                }
                                                            }
                                                        )
                                                    )
                                                )
                                            }
                                        )
                                    ]
                                )
                            )
                        }
                    )
                , WithProv
                    ( Derived desugar
                        ( Source
                            ( Parsed
                                ( Location
                                    { file = "test/golden/duplicated_signature.sonic"
                                    , span = Span
                                        { begin = Position
                                            { line = Line 2
                                            , column = Column 1
                                            }
                                        , end = Position
                                            { line = Line 2
                                            , column = Column 7
                                            }
                                        }
                                    }
                                )
                            )
                        )
                    )
                    ( Bind
                        { attrs = WithProv
                            ( Source ( Generated desugar ) )
                            ( Attrs ( fromList [] ) )
                        , id = WithProv
                            ( Source ( Generated desugar ) )
                            ( Unique ( MkUnique 0 ) )
                        , type_ = Nothing
                        , rhs = WithProv
                            ( Source
                                ( Parsed
                                    ( Location
                                        { file = "test/golden/duplicated_signature.sonic"
                                        , span = Span
                                            { begin = Position
                                                { line = Line 2
                                                , column = Column 5
                                                }
                                            , end = Position
                                                { line = Line 2
                                                , column = Column 7
                                                }
                                            }
                                        }
                                    )
                                )
                            )
                            ( Literal ()
                                ( WithProv
                                    ( Source
                                        ( Parsed
                                            ( Location
                                                { file = "test/golden/duplicated_signature.sonic"
                                                , span = Span
                                                    { begin = Position
                                                        { line = Line 2
                                                        , column = Column 5
                                                        }
                                                    , end = Position
                                                        { line = Line 2
                                                        , column = Column 7
                                                        }
                                                    }
                                                }
                                            )
                                        )
                                    ) ( Integer 1 )
                                )
                            )
                        }
                    )
                ]
            )
        ]
    , dataDecls = []
    , classDecls = []
    , instanceDecls = []
    , extension = ()
    }
, [ TestReport Error "DuplicatedTypeSignature {signatureDeclProvs = [Source (Parsed (Location {file = "test/golden/duplicated_signature.sonic", span = Span {begin = Position {line = Line 4, column = Column 1}, end = Position {line = Line 5, column = Column 1}}})),Derived desugar (Derived desugar (Source (Parsed (Location {file = "test/golden/duplicated_signature.sonic", span = Span {begin = Position {line = Line 2, column = Column 1}, end = Position {line = Line 2, column = Column 7}}}))))], declaredName = Name "a"}" ]
)